patches/api/0449-Split-Position-from-Location-type-hierarchy.patch

```diff
@@ -0,0 +1,126 @@
+         }
+ 
+         @Deprecated(forRemoval = true)
+diff --git a/src/main/java/org/bukkit/Location.java b/src/main/java/org/bukkit/Location.java
+index 9bbd928f7d513ca317cd27beffa61e5111f5ffb0..3ba409ce61e6ccf1ddfb961884454005fa0c842b 100644
+--- a/src/main/java/org/bukkit/Location.java
++++ b/src/main/java/org/bukkit/Location.java
+@@ -30,7 +30,7 @@ import org.bukkit.entity.Player;
+  * magnitude than 360 are valid, but may be normalized to any other equivalent
+  * representation by the implementation.
+  */
+-public class Location implements Cloneable, ConfigurationSerializable, io.papermc.paper.math.FinePosition { // Paper
++public class Location implements Cloneable, ConfigurationSerializable/*, io.papermc.paper.math.FinePosition*/ { // Paper
+     private Reference<World> world;
+     private double x;
+     private double y;
+@@ -120,7 +120,7 @@ public class Location implements Cloneable, ConfigurationSerializable, io.paperm
+      */
+     @NotNull
+     public Chunk getChunk() {
+-        return getWorld().getChunkAt(this);
++        return getWorld().getChunkAt(this.asPosition());
+     }
+ 
+     /**
+@@ -130,7 +130,7 @@ public class Location implements Cloneable, ConfigurationSerializable, io.paperm
+      */
+     @NotNull
+     public Block getBlock() {
+-        return getWorld().getBlockAt(this);
++        return getWorld().getBlockAt(this.asPosition());
+     }
+ 
+     /**
+@@ -1172,29 +1172,8 @@ public class Location implements Cloneable, ConfigurationSerializable, io.paperm
+     }
+ 
+     // Paper - add Position
+-    @Override
+-    public double x() {
+-        return this.getX();
+-    }
+-
+-    @Override
+-    public double y() {
+-        return this.getY();
+-    }
+-
+-    @Override
+-    public double z() {
+-        return this.getZ();
+-    }
+-
+-    @Override
+-    public boolean isFinite() {
+-        return io.papermc.paper.math.FinePosition.super.isFinite() && Float.isFinite(this.getYaw()) && Float.isFinite(this.getPitch());
+-    }
+-
+-    @Override
+-    public @NotNull Location toLocation(@NotNull World world) {
+-        return new Location(world, this.x(), this.y(), this.z(), this.getYaw(), this.getPitch());
++    public io.papermc.paper.math.Position asPosition() {
++        return io.papermc.paper.math.Position.fine(this);
+     }
+     // Paper end
+ }
+diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
+index 567a36a4887da8994c9170e2885aa8cc357efa0c..f678d4d3b91a8ab592580284bb9ceccfb4551b89 100644
+--- a/src/main/java/org/bukkit/World.java
++++ b/src/main/java/org/bukkit/World.java
+@@ -119,7 +119,7 @@ public interface World extends RegionAccessor, WorldInfo, PluginMessageRecipient
+      * @return Block at the given location
+      */
+     @NotNull
+-    public Block getBlockAt(@NotNull Location location);
++    public Block getBlockAt(@NotNull io.papermc.paper.math.Position location);
```

Comment:
all the param names should be adjusted too, or did you skip them to minimize diff (since you need to touch javadocs too)
also, some of them are paper methods, is it worth folding that back to the OG patch? or do we wait till softspoon?

Reply:
I just chose some random methods to demo this for. =

We could start by just doing all methods that paper adds, but we really want to do it in large chunks so that there isn't a point where half the codebase is using one and not the other. That gets annoying real quick.